low-level/framework/utils/iem.py

```diff
@@ -155,6 +159,9 @@ def check_exsisting_fault_iems(self):
             if prev_fault_iem_event:
                 self.fault_iems.append(event_name)
 
-def generate_iem(iem):
-    """Generate iem."""
-    syslog.syslog(iem)
+    def generate_iem(self, module, event_code, severity, description, specific_info):
+        """Generate iem and send it to a MessgaeBroker."""
+
+        EventMessage.send(module=module, event_id=event_code,
```

Comment:
no exception handling? We need to have it, even if EventMessage is not raising explicitly any !

Reply:
Done